### PR TITLE
Add function and tests for recursive immutability

### PR DESCRIPTION
**Pull request by eapache**
@boourns @lemonmade @bouk @carsonb 

Requires initial-state to be completely immutable, not just immutable at the top level.

My JS-typing is not strong, so hopefully I got the logic right.

### CODE REVIEW

File: src/createStore.js [hunk 1]
```diff
@@ -20,10 +20,25 @@ export type StoreType = {
   [key:string]: Function
 };
 
+function isRecursivelyImmutable(x: Immutable.Iterable): boolean {
+  let type = typeof x;
+  if (type === 'undefined' || x === null) {
```
Comment (lemonmade) [inline src/createStore.js]: You can do `type == null` for an undefined/ null test.

**Comment by lemonmade** [comment]
Change LGTM, could maybe use more tests for accepting primitives/ rejecting custom class instances and functions.

File: src/createStore.js (hunk 1 above)
Comment (eapache) [inline src/createStore.js]: `let x = null; typeof x` gives me 'object', not null?

**Comment by eapache** [comment]
More improvements will come later I think, this is good enough for now.

File: src/createStore.js (hunk 1 above)
Comment (lemonmade) [inline src/createStore.js]: ==, not single or triple equals (I know, JS sucks :P)

**Pull request merged**